### PR TITLE
api,server: add params for updatehypervisorcapabilities API

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Allows updating following capabilities for a hypervisor, version:
- Max DATA volumes limit
- Storage motion supported
- Max hosts per cluster
- VM snapshot enabled

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?
Using cmk:

```
(local) 🦌 > list hypervisorcapabilities id=7e83442b-19ed-11ec-93bc-645d8651f45a
{
  "count": 1,
  "hypervisorCapabilities": [
    {
      "hypervisor": "Simulator",
      "hypervisorversion": "default",
      "id": "7e83442b-19ed-11ec-93bc-645d8651f45a",
      "maxdatavolumeslimit": 6,
      "maxguestslimit": 50,
      "securitygroupenabled": true,
      "storagemotionenabled": true,
      "vmsnapshotenabled": false
    }
  ]
}
(local) 🦅 > update hypervisorcapabilities id=7e83442b-19ed-11ec-93bc-645d8651f45a maxdatavolumeslimit=32 maxguestslimit=64 securitygroupenabled=false storagemotionenabled=false vmsnapshotenabled=true maxhostspercluster=10
{
  "null": {
    "hypervisor": "Simulator",
    "hypervisorversion": "default",
    "id": "7e83442b-19ed-11ec-93bc-645d8651f45a",
    "maxdatavolumeslimit": 32,
    "maxguestslimit": 64,
    "maxhostspercluster": 10,
    "securitygroupenabled": false,
    "storagemotionenabled": false,
    "vmsnapshotenabled": true
  }
}
```

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by sureshanaparti** [comment]
@shwstppr Is this PR ready?

**Review by DaanHoogland — COMMENTED** [review]
code looks generally good but I think in this case a simple integration test is in order.
a simple
`update` -> `list` -> `verify` -> `update` -> `list` -> `verify`
what do you think @shwstppr ?

**Comment by shwstppr** [comment]
@DaanHoogland can you please have a look at the integration test added in the last commit. Should it be under components or smoke?

File: test/integration/smoke/test_hypervisor_capabilities.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+        vm_snapshot = VmSnapshot.create(
+            self.userapiclient,
+            vmid=vm.id,
+            snapshotmemory="false",
+            name="Test Snapshot",
+            description="Test Snapshot Desc"
+        )
+        if vm_snapshot:
+            VmSnapshot.deleteVMSnapshot(self.userapiclient, vm_snapshot.id)
```
Comment (DaanHoogland) [inline test/integration/smoke/test_hypervisor_capabilities.py:276]:
```suggestion
        self.cleanup.append(vm_snapshot)
```

**Review by DaanHoogland — COMMENTED** [review]
thanks for the test @shwstppr , some code suggestions and I see one issue in the test. Can you explain please?

File: test/integration/smoke/test_hypervisor_capabilities.py (hunk 1 above)
Comment (shwstppr) [reply test/integration/smoke/test_hypervisor_capabilities.py:273]: @DaanHoogland this doesn't work

Comment (shwstppr) [reply test/integration/smoke/test_hypervisor_capabilities.py]: had to make change to `lib/base.py`

File: test/integration/smoke/test_hypervisor_capabilities.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        try:
+            fail_snapshot = VmSnapshot.create(
+                self.userapiclient,
+                vmid=vm.id,
+                snapshotmemory="false",
+                name="Test Snapshot",
+                description="Test Snapshot Desc"
+            )
+            if vm_snapshot:
+                VmSnapshot.deleteVMSnapshot(self.userapiclient, fail_snapshot.id)
```
[the remark this answers is not in the document]
Comment (shwstppr) [reply test/integration/smoke/test_hypervisor_capabilities.py]: my bad

File: tools/marvin/marvin/lib/base.py [hunk 3]
```diff
@@ -5097,6 +5097,11 @@ def deleteVMSnapshot(cls, apiclient, vmsnapshotid):
         cmd.vmsnapshotid = vmsnapshotid
         return apiclient.deleteVMSnapshot(cmd)
 
+    def delete(self, apiclient):
```
Comment (DaanHoogland) [inline tools/marvin/marvin/lib/base.py:5100]: :+1:

**Review by DaanHoogland — APPROVED** [review]
clgtm

File: test/integration/smoke/test_hypervisor_capabilities.py (hunk 1 above)
Comment (DaanHoogland) [reply test/integration/smoke/test_hypervisor_capabilities.py]: ok, hadn;t realised that, but good solution :+1:

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@shwstppr , these trillian failures seem related to the changes, can you have a look?

**Review by borisstoyanov — APPROVED** [review]
except the failing tests, rest LGTM

**Comment by shwstppr** [comment]
@DaanHoogland will check and update

**Comment by shwstppr** [comment]
@DaanHoogland as discussed, moved test to component and made fixes.
- Detach volume was failing intermittently for multiple disks so added a stop VM statement
- VM snapshots for a running VM is not supported on KVM so skipped that test for KVM.

Tested VMware and KVM. KVM logs below:

```
[root@pr5473-t3104-kvm-centos7-marvin marvin]# nosetests --with-xunit --xunit-file=results.xml --with-marvin --marvin-config=./pr5473-t3104-kvm-centos7-advanced-cfg -s -a tags=advanced --hypervisor=KVM tests/component/test_hypervisor_capabilities.py 

==== Marvin Init Started ====

=== Marvin Parse Config Successful ===

=== Marvin Setting TestData Successful===

==== Log Folder Path: /marvin/MarvinLogs/Feb_02_2022_06_34_09_97PJFK All logs will be available here ====

=== Marvin Init Logging Successful===

==== Marvin Init Successful ====
=== TestName: test_01_check_hypervisor_max_data_volume_limit | Status : SUCCESS ===

=== Final results are now copied to: /marvin//MarvinLogs/test_hypervisor_capabilities_1R6CDP ===
[root@pr5473-t3104-kvm-centos7-marvin marvin]# cat /marvin//MarvinLogs/test_hypervisor_capabilities_1R6CDP/results.txt 
Test hypervisor maxdatavolumeslimit effect ... === TestName: test_01_check_hypervisor_max_data_volume_limit | Status : SUCCESS ===
ok
Test hypervisor vmsnapshotenabled effect ... SKIP: Skipping test: Reason -  VM Snapshot of running VM is not supported for KVM

----------------------------------------------------------------------
Ran 2 tests in 90.896s

OK (SKIP=1)
```

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Pull request merged**